src/backend/services/metrics.py

```diff
@@ -256,38 +234,27 @@ def log_signal_curl(signal: MetricsSignal) -> None:
             f"\n\ncurl -X POST -H \"Content-Type: application/json\" -d '{json_signal}' $ENDPOINT\n\n"
         )
 
-    @staticmethod
-    def get_agent(request: Request) -> Union[MetricsAgent, None]:
-        if not hasattr(request.state, "agent") or not request.state.agent:
-            return None
-        return request.state.agent
-
 
 # DO NOT THROW EXPCEPTIONS IN THIS FUNCTION
 class ChatMetricHelper:
     @staticmethod
-    def report_streaming_chat_event(event: dict[str, Any], **kwargs: Any) -> None:
+    def report_streaming_chat_event(
+        event: dict[str, Any], ctx: Context, **kwargs: Any
+    ) -> None:
+        start_time = None
         try:
-            request = kwargs.get("request", None)
-            if not request:
-                raise ValueError("request not set")
             event_type = event["event_type"]
             if event_type == StreamEvent.STREAM_START:
-                request.state.stream_start = time.perf_counter()
```

Comment:
removing this will likely break the duration feature, the `start_time` variable is destroyed as the function exits and not persisted across the duration of the stream

Reply:
One solution can be to track stream start in the context, and then pulling it out to calcualte duration ms, similar to how we were leveraging `request.state`

Reply:
Oh I didn't know about the variable was destroyed 😅  good catch, I'll add it to the context!